src/gui/app.cpp

```diff
@@ -1,18 +1,150 @@
 #include "app.h"
 
+#include "arguments.h"
+#include "mainwindow.h"
+#include "printinfo.h"
+#include "version.h"
+
 #include <QDir>
 #include <QFileInfo>
 #include <QFileOpenEvent>
 #include <QSettings>
 #include <QTextStream>
-
-#include "arguments.h"
-#include "mainwindow.h"
-#include "printinfo.h"
-#include "version.h"
+#include <vector>
 
 namespace NeovimQt {
 
+MainWindow* s_lastActiveWindow{ nullptr };
+std::vector<MainWindow*> s_windows;
+int s_exitStatus{ 0 };
+
+struct ConnectorInitArgs
+{
+	enum class Type
+	{
+		Embed,
+		Server,
+		Spawn,
+		Default
+	};
+
+	ConnectorInitArgs(const QCommandLineParser& parser, QStringList nvimArgs)
+		: type{ getConnectorType(parser) }
+		, timeout{ parser.value("timeout").toInt() }
+		, server{ parser.value("server") }
+		, nvim{ parser.value("nvim") }
+		, positionalArgs{ parser.positionalArguments() }
+		, neovimArgs{ std::move(nvimArgs) }
```

Comment:
Are the move semantics here intentional?

You're copying for the param, and then moving the copied param.

I'd recommend:
1. `const QStringList nvimArgs` and get rid of the `std::move`.
2. Use `QStringList&&` to move from the caller. (better if possible).